### ISSUE TITLE
Philosophy to deal with nan values in y data

### ISSUE THREAD

**Issue by BenjaminDAnjou** [f0uriest/interpax#160]
Hello there, love the library, a very welcome addition. However, I noticed that, unlike some other libraries I've seen out there, `interpax` doesn't ignore `nan` values in the input data. For instance, the following code

```python
import jax.numpy as jnp
from interpax import Interpolator1D
xs = jnp.asarray([1.,2.,3.,4.,5.,6])
ys = jnp.asarray([1.,4.,9.,jnp.nan,25.,36.])
interp = Interpolator1D(xs, ys)
result = interp(3.2)
```
outputs `nan` as a result. I also don't see a discussion of this in the documentation.

I was wondering if there were plans to include something like this, or why this choice was made, or how it is best to deal with it within `interpax`.

I understand that conceptually, it might be easier to assume that the data is pre-processed to exclude `nan` values.

**Comment by f0uriest** [comment]
Can you point me to how this is handled in other libraries? In scipy they generally just raise an error on invalid input, and we match that for the scipy-like classes eg `interpax.CubicSpline`. I suppose we could do the same for the `Interpolator` classes?

As for just ignoring the invalid values I don't think that would be easy with jax. Because of the need for static array sizes we can't just delete invalid elements, we'd need to mask them out and then plumb that masking all the way down.

My general philosphy is, as a library we try to ensure that we don't *generate* any nans for valid input data, but if the input is nan the least bad options are either throwing an error (tricky in jax) or pass the nans through, rather than trying to paste over them and potentially hiding a bigger issue upstream.
